provider/provider.go

```diff
@@ -226,3 +223,23 @@ func (dc DependencyCondition) Evaluate(ctx context.Context, log logr.Logger, con
 
 	return resp, nil
 }
+
+func getVersion(depVersion string) (*version.Version, error) {
+	v, err := version.NewVersion(depVersion)
+	if err == nil {
+		return v, nil
+	}
+	// Parsing failed so we'll try to extract a version and parse that
```

Comment:
for my edification, what are the cases this happens?

Reply:
Just in case you find it useful, this is what windup uses to compare versions: https://github.com/apache/maven/blob/master/maven-artifact/src/main/java/org/apache/maven/artifact/versioning/ComparableVersion.java#L538

used in here: https://github.com/windup/windup/blob/master/rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/condition/Version.java#L56

Reply:
yeah will do as a followup